docs/graphx-programming-guide.md

```diff
@@ -708,7 +708,9 @@ messages remaining.
 > messaging function.  These constraints allow additional optimization within GraphX.
 
 The following is the type signature of the [Pregel operator][GraphOps.pregel] as well as a *sketch*
-of its implementation (note calls to graph.cache have been removed):
+of its implementation (note: to avoid stackOverflowError due to long lineage chains, graph and 
```

Comment:
OK. Have removed references to checkpoint in the sketch and documented the config property in the spark configuration document in the latest update.